packages/ui-default/components/contestProblemEditor/ContestProblemEditor.tsx

```
@@ -0,0 +1,200 @@
+const randomId = () => Math.random().toString(16).substring(2);
+const ContestProblemEditor: React.FC<ContestProblemEditorProps> = ({ problems: initialProblems, onChange: _onChange }) => {
+  // TODO: also support balloon and other fields in the future
+  const [problems, setProblems] = React.useState<Problem[]>(initialProblems.map((el) => ({ ...el, _tmpId: randomId() })));
+
+  const problemRefs = React.useRef<{ [key: number]: any }>({});
+  const [problemRawTitles, setProblemRawTitles] = React.useState<Record<number, string>>({});
+
+  const fetchProblemTitles = debounce(async (ids: number[]) => {
+    api('problems', { ids }, ['docId', 'pid', 'title'])
+      .then((res) => {
+        setProblemRawTitles(res.reduce((acc, cur) => ({ ...acc, [cur.docId]: cur.title }), {}));
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)